odo/tests/test_convert.py

```diff
@@ -228,7 +229,15 @@ def test_pandas_and_chunks_pandas():
     assert len(list(c)) == 2
 
     df2 = chunks_dataframe_to_dataframe(c)
-    assert str(df2) == str(df)
+    tm.assert_frame_equal(df, df2)
+
+
+def test_iterator_to_DataFrame_chunks():
+    data = ((0, 1), (2, 3), (4, 5), (6, 7))
+    df1 = pd.DataFrame(list(data))
+    df2 = iterator_to_DataFrame_chunks(data, 2)
```

Comment:
I added a black box, but I couldn't (in about 30 seconds of trying) convince myself that this would get all the keywords right, so I did it both ways.